airbyte-workers/src/main/java/io/airbyte/workers/internal/EmptyAirbyteSource.java

```diff
@@ -5,29 +5,88 @@
 package io.airbyte.workers.internal;
 
 import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.ResetSourceConfiguration;
+import io.airbyte.config.StateType;
+import io.airbyte.config.StateWrapper;
+import io.airbyte.config.StreamDescriptor;
 import io.airbyte.config.WorkerSourceConfig;
+import io.airbyte.config.helpers.StateMessageHelper;
+import io.airbyte.protocol.models.AirbyteGlobalState;
 import io.airbyte.protocol.models.AirbyteMessage;
 import io.airbyte.protocol.models.AirbyteMessage.Type;
 import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.AirbyteStreamState;
 import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedList;
 import java.util.Optional;
+import java.util.Queue;
+import java.util.Set;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
 
 /**
  * This source will never emit any messages. It can be used in cases where that is helpful (hint:
  * reset connection jobs).
  */
+@Slf4j
 public class EmptyAirbyteSource implements AirbyteSource {
 
   private final AtomicBoolean hasEmittedState;
+  private final Queue<StreamDescriptor> streamsToReset = new LinkedList<>();
+  private boolean isResetBasedOnConfig;
+  private boolean isStarted = false;
+  private Optional<StateWrapper> stateWrapper;
 
   public EmptyAirbyteSource() {
     hasEmittedState = new AtomicBoolean();
   }
 
   @Override
   public void start(final WorkerSourceConfig sourceConfig, final Path jobRoot) throws Exception {
-    // no op.
+
+    if (sourceConfig == null || sourceConfig.getSourceConnectionConfiguration() == null) {
+      // TODO: When the jobConfig is fully updated and tested, we can remove this extra check that makes
+      // us compatible with running a reset with
+      // a null config
+      isResetBasedOnConfig = false;
+    } else {
+      final ResetSourceConfiguration sourceConfiguration;
```

Comment:
It's a bit confusing to me that we have both `sourceConfig` and `sourceConfiguration` as variable names in this method. Could you rename these to be more explicit, e.g. `workerSourceConfig` and `resetSourceConfiguration`?

Reply:
Done,